From 3f507316aeaad7c667fdbc388a692f04948a96a7 Mon Sep 17 00:00:00 2001
From: Alberto Guanti <albertoguanti@gmail.com>
Date: Mon, 20 Mar 2017 23:12:45 +0100
Subject: [PATCH] DO NOT MERGE "[PATCH] Zygote: Stop breaking the entire
 system"

* This is just an hack required to boot the system until a proper fix to
this error is found.

11-17 22:34:48.294  2745  2745 E Zygote  : Unsupported st_mode 384
11-17 22:34:48.295  2745  2745 F art     :
art/runtime/jni_internal.cc:497] JNI FatalError called:
frameworks/base/core/jni/com_android_internal_os_Zygote.cpp:478: Unable
to construct file descriptor table.

Change-Id: Ic295de39f1a5dc821a36b111d08916c7ed9bcb37
---
 core/jni/com_android_internal_os_Zygote.cpp | 3 ++-
 2 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/core/jni/com_android_internal_os_Zygote.cpp b/core/jni/com_android_internal_os_Zygote.cpp
index 6118d1b36b9..6fdafb0a39e 100644
--- a/core/jni/com_android_internal_os_Zygote.cpp
+++ b/core/jni/com_android_internal_os_Zygote.cpp
@@ -520,7 +520,8 @@ static pid_t ForkAndSpecializeCommon(JNIEnv* env, uid_t uid, gid_t gid, jintArra
     // Re-open all remaining open file descriptors so that they aren't shared
     // with the zygote across a fork.
     if (!gOpenFdTable->ReopenOrDetach()) {
-      RuntimeAbort(env, __LINE__, "Unable to reopen whitelisted descriptors.");
+      //RuntimeAbort(env, __LINE__, "Unable to reopen whitelisted descriptors.");
+      ALOGE("FIXME: Unable to reopen whitelisted descriptors.");
     }
 
     if (sigprocmask(SIG_UNBLOCK, &sigchld, nullptr) == -1) {

